### PR TITLE
Correcly handle CRLF in the manifest

### PR DESCRIPTION
**Pull request by gtrafimenkov**
kops replace/create/delete now correctly handle crlf in the manifest.

This fixes issue #6532

### CODE REVIEW

**Comment by chrisz100** [comment]
/ok-to-test 

Can you please update the bazel files as you made changes there? the CI won't pass this otherwise. 
`FAIL: ./hack/verify-bazel.sh failed, as the bazel files are not up to date
FAIL: Please execute the following command: ./hack/update-bazel.sh
make: *** [verify-bazel] Error 1
The command "make ci" exited with 2.`

**Comment by gtrafimenkov** [comment]
Thank you, @chrisz100.  I've done necessary changes.

File: cmd/kops/create.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ func RunCreate(f *util.Factory, out io.Writer, c *CreateOptions) error {
 			}
 		}
 		// TODO: this does not support a JSON array
-		sections := bytes.Split(bytes.Replace(contents, []byte("\r\n"), []byte("\n"), -1), []byte("\n---\n"))
+		sections := text.SplitContentToSections(contents)
```
Comment (justinsb) [inline cmd/kops/create.go]: This is the perfect name for the function - I was wondering "how did I now know about that function" as it looked so much like a k8s function :-)

**Comment by justinsb** [comment]
Thanks @gtrafimenkov - this is a big improvement.  I do worry that we might be replacing `\r` with `\n` in some values (not just the separators), so maybe we should technically split on all of `\r\n---\r\n`, `\n---\r\n`, `\r\n---\n` and `\n---\n`.  We could also try splitting on a regex or function, but that's probably more complicated.

But .. I don't think we have anything where `\r\n` will matter in a value, so going to approve!

/approve
/value

**Comment by justinsb** [comment]
Looks like I forgot the lgtm - sorry!

/lgtm

**Pull request merged**